File: ThreadingSimulator/Models/ExecutableProgramModel.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ThreadingSimulator.Models
{
    public class ExecutableProgramModel
    {
        public string Name { get; set; }
        public List<ProcessCommandsModel> Processes { get; set; }
        
        public int ProcessCount
        {
            get
            {
                return Processes != null ? Processes.Count : -1;
            }
        }
        
        public void Add(ProcessCommandsModel commands)
        {
            if(Processes==null)
            {
                Processes = new List<ProcessCommandsModel>();
            }

            Processes.Add(commands);
        }
    }
}

File: GUI/src/app/Models/ExecutableProgramModel.ts
import { ProcessCommandsModel } from "./ProcessCommandsModel";

export class ExecutableProgramModel
{
    public Name: string;
    public Processes: ProcessCommandsModel[];
    
    public ProcessCount(): number
    {
        return this.Processes != null ? this.Processes.length : -1;
    }
    
    public Add(commands: ProcessCommandsModel)
    {
        if(this.Processes==null)
        {
            this.Processes = [];
        }

        this.Processes.push(commands);
    }

    public static Deserialize(obj: any): ExecutableProgramModel
    {
        let trueObj: ExecutableProgramModel = new ExecutableProgramModel();

        trueObj.Name = obj.Name;
        
        if(obj.Processes)
        {
            trueObj.Processes = [];

            obj.Processes.forEach(element => {
                trueObj.Processes.push(ProcessCommandsModel.Deserialize(element));
            });
        }

        return trueObj;
    }
}